modules/tinymce/src/plugins/table/main/ts/ui/TableDialog.ts

```diff
@@ -48,6 +48,7 @@ const applyDataToElement = (editor: Editor, tableElm: HTMLTableElement, data: Ta
 
   const shouldStyleWithCss = Options.shouldStyleWithCss(editor);
   const hasAdvancedTableTab = Options.hasAdvancedTableTab(editor);
+  const borderIsZero = data.border?.includes?.('0');
```

Comment:
Wouldn't this return true when `data.border` has a value of '10' or has another 0 in it?
We should be able to use `parseFloat` which should automatically strip any non-number characters like 'px'
```suggestion
  const borderIsZero = parseFloat(data.border) === 0;
```